### PR TITLE
New "bugfix" tag for addons

### PR DESCRIPTION
**Pull request by Norbiros**
Resolves https://github.com/ScratchAddons/ScratchAddons/pull/2568#issuecomment-876833246

### Changes

Adds new tag to addons: `don't run duplicated blocks` and `fix SVG upload`
file

### Tests

Firefox newest.

### CODE REVIEW

File: _locales/en/messages.json [hunk 1]
```diff
@@ -188,6 +191,9 @@
   "betaTooltip": {
     "message": "Keep in mind that this addon might have bugs."
   },
+  "bugfixTooltip": {
+    "message": "This addon fix Scratch's bugs!"
```
Comment (Samq64) [inline _locales/en/messages.json:195]:
```suggestion
    "message": "This addon fix Scratch's bugs!"
```
    "message": "This addon fixes a Scratch bug."

**Comment by Norbiros** [comment]
Okay, added.

------- Original Message -------
wtorek, 19 kwietnia 2022 14:40, Samq64 ***@***.***> napisał(a):

**Comment by Samq64** [comment]
> Okay, added.

Sorry, I didn't mean to post that.

**Comment by RedGuy12** [comment]
> > Okay, added.
> 
> I didn't mean to post that review.

it does sound better tho tbh

File: webpages/settings/data/addon-groups.js [hunk 2]
```diff
@@ -92,4 +92,12 @@ export default [
...
     fullscreenShow: true,
   },
+  {
+    id: "bugfix",
+    name: "bugfix",
+    addonIds: [],
+    expanded: false,
+    iframeShow: false,
+    fullscreenShow: true,
+  },
```
Comment (WorldLanguages) [inline webpages/settings/data/addon-groups.js:102]: Adding a group here, without modifying index.js, won't do anything.

File: addons/block-duplicate/addon.json [hunk 3]
```diff
@@ -22,6 +22,6 @@
   "dynamicEnable": true,
   "dynamicDisable": true,
   "versionAdded": "1.21.0",
-  "tags": ["editor", "codeEditor", "featured"],
+  "tags": ["editor", "codeEditor", "featured", "bugfix"],
```
Comment (WorldLanguages) [inline addons/block-duplicate/addon.json:25]: I think you meant to add this to `fix-pasted-scripts` instead.

**Comment by WorldLanguages** [comment]
@mxmou @lisa-wolfgang What do you think about this PR?

I'm unsure. We don't allow users to list/filter addons with a tag, so should we create an addon group dedicated to addons that fix Scratch bugs? Enabled, recommended, bugfixes(?), featured, forums, others, beta?
How can we explain "this addon fixes a Scratch bug" with fewer words? Can't "bug fix" be confused with an addon we made that had a bug, then we fixed our bug?

**Comment by lisa-wolfgang** [comment]
I like the idea of having an addon group without showing the tag on each addon, like what we do for featured addons. Like W_L said, putting the tag directly on the addon in a similar manner as the "updated" tag implies that we're fixing our own bugs, but a group has a more permanent connotation, especially since it would be distanced from the "new addons and updates" sections.

However, what I'd like even more is for these addons to be enabled by default. I know that it's been a challenge to come up with a universal solution for indicating our modifications, but I don't think that's the best way to think about it. Every addon is different and will need its own indication, so such a system will need to be flexible. Therefore, I'd suggest the following implementation:
- `fix-pasted-scripts`: the first time the user performs a duplicate or paste action, show a popup near the code editor that says something to the effect of "hey, we made it so that putting that block on another one doesn't run them anymore."
- `fix-uploaded-svgs`: the first time the user uploads an SVG file, show a popup near the costume editor that says something to the effect of "hey, we made it so that SVG files upload properly."
- `fix-editor-comments`: since the user may have been deterred from using comments in Scratch at all, the first time the user creates a block, show a popup near the code editor that says something to the effect of "hey, we made it so that comments actually move with blocks they're attached to."

File: webpages/settings/data/addon-groups.js (hunk 2 above)
Comment (Norbiros) [reply webpages/settings/data/addon-groups.js:102]: I know, but now this `Bugfix` tag is just indicator - it don't have specific group in settings page etc. But in the future, when someone wants to change it, the basics will be ready!

**Comment by RedGuy12** [comment]
> How can we explain "this addon fixes a Scratch bug" with fewer words? Can't "bug fix" be confused with an addon we made that had a bug, then we fixed our bug?

I think the tag should be called "bug fix" with a tooltip "this addon fixes a Scratch bug"

**Comment by lisa-wolfgang** [comment]
> > How can we explain "this addon fixes a Scratch bug" with fewer words? Can't "bug fix" be confused with an addon we made that had a bug, then we fixed our bug?
> 
> 
> 
> I think the tag should be called "bug fix" with a tooltip "this addon fixes a Scratch bug"

That doesn't solve the issue of being confusable with our own bugs. Tooltips aren't reliable enough.

**Comment by WorldLanguages** [comment]
> However, what I'd like even more is for these addons to be enabled by default.

I think discussing that is out of scope for this PR, maybe we can create an issue?

**Comment by lisa-wolfgang** [comment]
If we go ahead with this PR, it would make sense to have a "feature" tag as well, since there's already a tag for themes.

**Comment by mxmou** [comment]
> If we go ahead with this PR, it would make sense to have a "feature" tag as well, since there's already a tag for themes.

I don't think that would be very useful - almost every addon would have the tag.

**Comment by WorldLanguages** [comment]
I think we should have an open discussion about the settings page as a whole and deal with this as well as #3955 at the same time 🤔

**Comment by RedGuy12** [comment]
There's more bug fixes


________________________________
From: Norbiros ***@***.***>
Sent: Tuesday, April 19, 2022 5:39:44 AM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] New "bugfix" tag for addons (PR #4515)


Resolves #2568 (comment)<https://github.com/ScratchAddons/ScratchAddons/pull/2568#issuecomment-876833246>

Changes

Adds new tag to addons: don't run duplicated blocks and fix SVG upload
[file]<https://user-images.githubusercontent.com/89837724/163985576-3b6c4428-27af-448f-8b95-d9d8905d489c.png>

Tests

Firefox newest.

Other

Merge before:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/ScratchAddons/ScratchAddons/pull/4515

Commit Summary

  *   07e723f<https://github.com/ScratchAddons/ScratchAddons/pull/4515/commits/07e723f781a8c0c6d35650d89b28ad7faab743c0> New bugfix tag!

File Changes

(5 files<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files>)

  *   M _locales/en/messages.json<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files#diff-4362c7f7032e9687a0a5910cadc127afbe8259b2b941de40dd4246c35b1446f0> (6)
  *   M addons/block-duplicate/addon.json<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files#diff-604d9340f0dbd1d9629fc256bf917e30850811a1aa456482c7661bd1ad0d0088> (2)
  *   M addons/fix-uploaded-svgs/addon.json<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files#diff-3893773c1feab20eddc19935f6b0df645ac6705a5fffddf2c5d43d516551aa67> (2)
  *   M webpages/settings/data/addon-groups.js<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files#diff-ada5a2a719af1a479d6f5d083315939036b2eeff160258ae9278a5eafa0c2117> (8)
  *   M webpages/settings/data/tags.js<https://github.com/ScratchAddons/ScratchAddons/pull/4515/files#diff-e0d5fbf593cf0a26ebc3dfa4c0d5e81cec8431a30d1c3ebccbee1d130378608d> (7)

Patch Links:

  *   https://github.com/ScratchAddons/ScratchAddons/pull/4515.patch
  *   https://github.com/ScratchAddons/ScratchAddons/pull/4515.diff

**Comment by Samq64** [comment]
Should this be closed in favour of #7242?

**Comment by WorldLanguages** [comment]
Okay, let's focus on that one instead.

**Pull request closed without merging**